src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/BerConverter.Windows.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+
+namespace System.DirectoryServices.Protocols
+{
+    public static partial class BerConverter
+    {
+        private static unsafe int DecodeBitStringHelper(ArrayList resultList, SafeBerHandle berElement)
+        {
+            int error;
+            // return a bitstring and its length
+            IntPtr ptrResult = IntPtr.Zero;
+            int length = 0;
+            error = LdapPal.BerScanfBitstring(berElement, "B", ref ptrResult, ref length);
+
+            if (!LdapPal.IsBerDecodeError(error))
+            {
+                byte[] byteArray = null;
+                if (ptrResult != IntPtr.Zero)
+                {
+                    byteArray = new byte[length];
+                    Marshal.Copy(ptrResult, byteArray, 0, length);
+                }
+                resultList.Add(byteArray);
+            }
+            else
+                Debug.WriteLine("ber_scanf for format character 'B' failed");
```

Comment:
Since you've moved this to a new file, please add the braces (which are required for the else since the if has them).

```suggestion
            {
                Debug.WriteLine("ber_scanf for format character 'B' failed");
            }
```